### ISSUE TITLE
[Snackar] Add option to control casing on SnackBar action text

### ISSUE THREAD

**Issue by szeflerben** [material-components/material-components-ios#6052]
## Feature Request/Suggestion

SnackBar action text is always uppercased. Can something like `uppercaseTitle` be added to the `MDCSnackbarMessageAction` similar to how it is handled on `MDCButton`?

---
#### Internal data
- Associated internal bug: [b/121268372](http://b/121268372)

**Comment by romoore** [comment]
You could use the `MDCSnackbarManagerDelegate` method `-willPresentSnackbarWithMessageView:` method to retrieve the view. From there, you could inspect the `actionButtons` array to find the button you want and set `uppercaseTitle = NO;`.  Would that work for your use case?

**Comment by szeflerben** [comment]
I see how that could work but I have noticed that other API's have been relaxed on other components to allow more control over casing ([here](https://github.com/material-components/material-components-ios/pull/4935)). I think that we would ideally prefer a similar solution the attached link. Would this be possible? If not we will go down the delegate route

**Comment by romoore** [comment]
@szeflerben I agree that a pass-through API would be simpler in this case. I was suggesting the delegate API as a mechanism to unblock your work at the moment. I think the request of a new API is very reasonable and will hopefully be put into our backlog to get worked on in 2019.  

In the mean time, however, I would not postpone any development hoping that it will get done quickly.  Once the API is available, switching to it should hopefully be a relatively small refactoring on your side.

**Comment by szeflerben** [comment]
That's perfect. Thanks

**Comment by adamirowe** [comment]
This functionality would be great for us as well.

**Comment by adamirowe** [comment]
> MDCSnackbarManagerDelegate

@romoore Could you provide an example on how to use the delegate in Swift 4.2? Attempting to extend the delegate is giving me compilation issues.

**Comment by adamirowe** [comment]
Looks like this property is not exposed in the api: https://material.io/develop/ios/components/snackbars/api-docs/Classes/MDCSnackbarManager.html#/c:objc(cs)MDCSnackbarManager(py)uppercaseButtonTitle

**Comment by randallli** [comment]
Sadly we are trying to remove the button's uppercaseTitle API in the future and I cannot in good conscience expand the snackbar's API surface to include more of this.

Example of how to use the delegate:
```
- (void)willPresentSnackbarWithMessageView:(MDCSnackbarMessageView *)messageView {
  NSMutableArray<MDCButton *> *buttons = messageView.actionButtons;
  for (MDCButton *button in buttons) {
    button.uppercaseTitle = NO;
 }

```
Also don't forget to set the delegate. 
`
MDCSnackbarManager.defaultManager.delegate = self;
`
like how [this example](https://github.com/material-components/material-components-ios/blob/develop/components/Snackbar/examples/SnackbarSimpleExample.m#L63
) does.
